tutorials/how_to/debug_using_pycharm.md

```diff
@@ -0,0 +1,133 @@
+# How to Debug Your AllenNLP Code
+
+Recently several people have asked how to debug allennlp code
+using PyCharm, so I set out to figure it out. It turns out that
+it's not terribly difficult. It's also easy to do it in VSCode.
+
+If you have a different preferred debugger,
+these instructions will probably work for you too.
+
+## Some Code To Debug
+
+We'll demonstrate the debugger using the Academic Paper Classifier model
+from our ["AllenNLP-as-a-Library" example](https://github.com/allenai/allennlp-as-a-library-example).
+
+If you'd like to follow along, clone that repo and install its requirements.
+
+## How to Debug in PyCharm, Using "Run > Attach to Local Process"
+
+Our recommended workflow is using our command-line tool `allennlp`.
+The example repo contains the training command:
+
+```
+allennlp train experiments/venue_classifier.json -s /tmp/your_output_dir_here --include-package my_library -o '{"trainer": {"cuda_device": -1}}'
+```
+
+(I added an override to train on the CPU, since the machine you're running PyCharm on probably doesn't have a GPU.)
+
+It turns out that `allennlp` is just (in essence) an alias for `python -m allennlp.run`, so you could equivalently do
+
+```
+python -m allennlp.run train experiments/venue_classifier.json -s /tmp/your_output_dir_here --include-package my_library -o '{"trainer": {"cuda_device": -1}}'
```

Comment:
What is your motivation for including this?  You might consider explaining why in the proceeding text.

Reply:
the motivation is that (as far as I can tell) the "attach to local process" will only find "python" processes, so if you do `allennlp train` you can't attach to it. I'll make that clearer